src/commons/degrade_cluster.py

```diff
@@ -66,13 +68,43 @@ def get_degraded_mode():
         os.environ['DEGRADE_POD'] = degrade_pod
 
 
-def activate_degraded_mode(options: munchify):
+def activate_degraded_mode_parallel(return_dict, m_conf):
     """Create bucket and degrade node as needed .
 
-    :param options : dictionary to fetch command line params
+    :param return_dict : dictionary to share data between parallel processes.
+    :param m_conf : dictionary to to fetch required info for parallel mode setup.
     :return : None
     """
+    return_dict.update({"degraded_done": False})
     get_degraded_mode()
+    for cnt in range(m_conf['degraded_io']['schedule_frequency']):
+        LOGGER.info("Degrading Cluster for %s count", cnt)
+        total_t = datetime.strptime(m_conf['degraded_io']['pod_downtime_schedule'],
+                                    '%H:%M:%S').time()
+        downtime = total_t.hour * 3600 + total_t.minute * 60 + total_t.second
+        LOGGER.info("Degraded Cluster Process Sleeping for %s sec before downtime.", downtime)
+        time.sleep(downtime)
+        return_dict.update({"is_deg_on": True})
+        logical_node = get_logical_node()
+        os.environ["logical_node"] = str(logical_node)
+        # Set default pod_prefix
+        pod_prefix = SERVER_POD_NAME_PREFIX
+        if os.environ['POD_PREFIX'].lower() == 'data':
+            pod_prefix = DATA_POD_NAME_PREFIX
+        LOGGER.info("***** Making Degraded Cluster *****")
+        logical_node.degrade_nodes(pod_prefix)
+        total_t = datetime.strptime(m_conf['degraded_io']['pod_uptime_schedule'], '%H:%M:%S').time()
+        uptime = total_t.hour * 3600 + total_t.minute * 60 + total_t.second
+        LOGGER.info("Degraded Cluster Process Sleeping for %s sec before uptime", uptime)
+        time.sleep(uptime)
+        LOGGER.info("Restoring From Degraded Cluster")
+        restore_pod()
```

Comment:
From function I can see we are bringing pod up after some interval. Can we make this conditional as some scenario might run in degraded mode only?

Reply:
> From function I can see we are bringing pod up after some interval. Can we make this conditional as some scenario might run in degraded mode only?

Taken care of the same